src/progressbar.rs

```diff
@@ -11,90 +12,26 @@ const PROGRESSBAR_HEIGHT: i32 = 6;
 #[derive(Shrinkwrap)]
 pub struct ProgressBar {
 	#[shrinkwrap(main_field)]
-	pub bar: gtk::DrawingArea,
+	pub bar: gtk::Box,
 }
 
 impl ProgressBar {
 	pub fn new(fraction: f64) -> Self {
-		let bar = gtk::DrawingArea::new();
-		bar.set_height_request(PROGRESSBAR_HEIGHT);
-
-		bar.connect_draw(move |area, ctx| {
-			let width = f64::from(area.allocated_width());
-			let height: f64 = f64::from(PROGRESSBAR_HEIGHT);
-			let radius: f64 = height * 0.5;
-
-			// Gets the themes bg and fg colors for the OSD window.
-			// Compares the fg and bg to determine if the theme is a
-			// dark or a light theme.
-			let def_fg = gdk::RGBA::new(0.0, 0.0, 0.0, 1.0);
-			let def_bg = gdk::RGBA::new(1.0, 1.0, 1.0, 1.0);
-			let (fg, bg, dark_mode) = match area.toplevel() {
-				Some(win) => {
-					let style = win.style_context();
-					let fg = style
-						.style_property_for_state(gtk::STYLE_PROPERTY_COLOR, StateFlags::NORMAL)
-						.get::<gdk::RGBA>();
-					let bg = style
-						.style_property_for_state(
-							gtk::STYLE_PROPERTY_BACKGROUND_COLOR,
-							StateFlags::NORMAL,
-						)
-						.get::<gdk::RGBA>();
-					let (fg, bg) = match (fg, bg) {
-						(Ok(fg), Ok(bg)) => (fg, bg),
-						_ => (def_fg, def_bg),
-					};
-					(fg, bg, is_dark_mode(&fg, &bg))
-				}
-				None => (def_fg, def_bg, is_dark_mode(&def_fg, &def_bg)),
-			};
-
-			// Background
-			if dark_mode {
-				ctx.set_source_rgb(fg.red() * 0.5, fg.green() * 0.5, fg.blue() * 0.5);
-			} else {
-				ctx.set_source_rgb(bg.red() * 0.75, bg.green() * 0.75, bg.blue() * 0.75);
-			}
-			ProgressBar::draw_rounded_rectangle(width, height, radius, ctx);
-
-			// Progress
-			ctx.set_source_rgb(fg.red(), fg.green(), fg.blue());
-			ProgressBar::draw_rounded_rectangle(width * fraction, height, radius, ctx);
```

Comment:
Hard coding the color values in the CSS can make it messy with third-party themes. The current code (in main) uses the themes colors, so still using that by default would be great.

Maybe we can keep this and use it to define a CSS variable on start that the default CSS style can use for the progress bar?

Would you be okay with me pushing changes to your branch?

Reply:
As I said in my comment, I haven't yet tackled making it support third-party themes because the core function does not work. The progressbar works on the first initialization, but it does not work from then on. Please look at my original comment.

Yes of course, feel free to push changes to my branch!

Reply:
Do I need to give you permissions on my repo, or can you push directly from here?

Reply:
> Do I need to give you permissions on my repo, or can you push directly from here?

I should be able to push without you changing anything :)

Reply:
Decided to use the themes default foreground color for the progressbar for it to be adaptive :)